### ISSUE TITLE
Hail is missing deploys 0.2.114 and 0.2.115 in Azure

### ISSUE THREAD

**Issue by danking** [hail-is/hail#13050]
### What happened?

https://hail.zulipchat.com/#narrow/stream/223457-Hail-Batch-support/topic/Hail.20batch.20on.20azure.20-.20remote_tmpdir.20error.3F

https://hail.zulipchat.com/#narrow/stream/300487-Hail-Batch-Dev/topic/0.2E2.2E115.20JAR.20missing.20in.20azure

https://batch.azure.hail.is/batches/3749715 seems to be the last one on april 19 and we pick up again on april 27 https://batch.azure.hail.is/batches/3751384

### Version

xxx

**Comment by jigold** [comment]
Just double checking -- I'm thinking I can dev deploy only the step `upload_query_jar` for those specific missing commits and then copy the resulting JAR to the appropriate place. Or can I build the JAR locally?

**Comment by danking** [comment]
You can just copy the JAR from GCS into ABS. The only special thing about Azure is that HDInsight relies on a rather old version of Spark. That has no bearing on QoB though.

**Comment by danking** [comment]
More concerning is why we just didn't deploy for a week

**Comment by jigold** [comment]
I thought @daniel-goldstein looked into this, but I can't find anything on Zulip. I'll ask him when he gets back if he remembers anything.

**Comment by daniel-goldstein** [comment]
> https://batch.azure.hail.is/batches/3749715 seems to be the last one on april 19 and we pick up again on april 27 https://batch.azure.hail.is/batches/3751384


I only went as far as to check the past deploys to find out this.

**Comment by jigold** [comment]
Known facts:
- the last CI k8s deployment that started the deploy=1 batch on April 27th was active since at least April 25th.
- For a small window of time that I looked at on April 25th, it kept getting errors when trying to get the Github status for possible merge candidates: 12848, 12849, 12547. There might be other PRs at later dates. I saw at least the same errors for 12848 on April 27th.

I'm going to throw out a hypothesis. I merged the dedup attempt resources PR on April 19th. The PRs that were stacked on previous commits of that PR now have merge conflicts with the set of commits that actually got merged. This caused problems because the next merge candidates CI was selecting was causing bad GitHub rate limit requests for exceeding the number of statuses. So it kept retrying that same merge candidate. CI didn't get restarted at least from the 25th to the 27th so the merge candidate never would have been refreshed. We know that there's less GKE node turnover in Azure, so not unexpected that the ci pod wouldn't get redeployed on its own. I'm thinking it's possible that I merged the database trigger fix on April 27th in response to the excessive deadlocks we noticed and then rebased the subsequent stacked PRs that had merge conflicts, thus unblocking CI, but I'm not sure (it's really hard to get what I want from the Azure log analytics system).

I think the "bug fix" here is to reassess the code in CI and possibly harden it where we select the merge candidate and try to get the status so it doesn't block deployments. I have a screenshot from April 25th below in case it's helpful. The log analytics query that is helpful is:

```
ContainerLog
| where ContainerID  == "273584134970cdae08cf0d412461862e2a0e558888a52c91870ca46a146cbb8a"
| order by TimeGenerated
```

<img width="1085" alt="Screen Shot 2023-05-24 at 12 58 18 PM" src="https://github.com/hail-is/hail/assets/1693348/e2da08b6-5982-46cb-9e2c-2178a19f2f86">

**Comment by daniel-goldstein** [comment]
Do you have an error message other than that failed post? I'm not seeing why that would stop a deploy. In #13115, the PR healing code is after the `_heal_deploy` code, so I don't know why an exception when healing a PR would stop a deploy from occurring. This POST error is also occurring on GCP. Definitely something to fix but I'm not sure why it's related.

**Comment by daniel-goldstein** [comment]
Alternate hypothesis that I'm not sure whether we can test: we forgot to unfreeze azure CI for a week?

**Comment by daniel-goldstein** [comment]
I searched [here](https://portal.azure.com#@d6c9f2ea-d3bb-4ca9-8b14-231bac999aa6/blade/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F22cd45fe-f996-4c51-af67-ef329d977519%2Fresourcegroups%2Fhaildev%2Fproviders%2Fmicrosoft.operationalinsights%2Fworkspaces%2Fhaildev-logs/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA21Sy27CMBC88xVbLiRSqNprUSpVgFpUVCraOzLxJhgcO7LNI2r7710nKRBoDokzmp2dnbVEB6%252FbJb5rPtWZhRiCQnPFciwMpuLwYJ0RKovAQ7ZgCR4RqTOJO5QNEMJXB%252BiRpDjUyjGh0Ez4VFhHqk2PidqhctqUFfUb9is0CG%252BkDdYx4%252BxeuBUdTcJc20gE3X43bJURTaIKTs1G4ePdlXBlGuL4NEFD4eRMqOTc7SgCHwRbooxOxRU4KWpkUFUfSyi0Rg4PDhUHGs%252BUccGMxcXaahUQY%252ByxsM0jbcsyjCv%252BbfPXMn9mC4SC4CLUdha1jKV9GOFKP%252B7fev6h4SFZCJVquElqTQvdj5enbkOVWm%252B2BWyE4rHE1OmtQ3PdH7Q699gUF0avMXF9tmclTHIaKqo%252Fn4zuzBwLbYW%252FAHWaEXyKHGfpUOc5U%252FxCw6DfmU%252Ffc2P%252Fuu%252F8DDpnFzboJaIXQY9jyrbS%252BeN4Pp%252FNeyHpaMPJ%252BLKsmjwj2WQOOXC0yS9DxlTe%252BAIAAA%253D%253D/timespan/2023-04-16T20%3A37%3A28.000Z%2F2023-04-28T20%3A37%3A28.000Z) and couldn't really find anything insightful. It does feel odd that we went that long with so *few* error messages, so maybe some silent error.

**Comment by jigold** [comment]
I think we must have forgotten to unfreeze Azure CI for a week. That would make sense with it correlating with two database migrations. Let me know if I can close this.

**Comment by jigold** [comment]
Found it! 
<img width="1400" alt="Screen Shot 2023-05-25 at 2 04 32 PM" src="https://github.com/hail-is/hail/assets/1693348/6271b24c-ee68-4374-9691-fcb5ab53f0fc">

**Comment by daniel-goldstein** [comment]
This is not because we forgot to unfreeze CI, we just have simply never added the dockerhub images to azure automatically. The couple that are there now (only 107 and 112) must have been uploaded manually.

Because there are some build.yaml steps that run on deploy that are specific to the broad GCP instance (like maybe making a release), non-hail-vdc instances don't run the whole build.yaml pipeline on deploy, but a subset that are specified through terraform (this is how AUS and MS could decide to only deploy a subset of our services e.g. not monitoring. We somewhat recently added a step (separate from the `deploy` step) called `mirror_hailgenetics_images` that was entirely intended so that other hail deployments (including ourselves on Azure!) could pick up the images that we released to dockerhub. I never added that steps to the Azure CI's config. I have done that now.

**Comment by daniel-goldstein** [comment]
Er, wait, sorry, this thread was about the JAR. That indeed should have been deployed under the current steps (because we run QoB tests). So this is a separate issue, but leaving my comment up for posterity

**Comment by danking** [comment]
Thanks for the clarity on the images and the fix.
